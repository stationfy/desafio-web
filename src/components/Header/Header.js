import React, { Component } from "react";
import { Link } from 'react-router-dom';

import './Header.css';

class Header extends Component {

  renderHeader(reponame) {
    if(!reponame) {
      return (
        <div className="header-title-block">
          <i className="fa fa-bars" aria-hidden="true" onClick={ () => { alert(`A menu here, soon...`) } }></i>
          <span className="header-title">Github JavaPop</span>
        </div>
      )
    } else {
      return (
        <div className="header-title-block">
          <Link to="/">
            <i className="fa fa-arrow-left header-back" aria-hidden="true"></i>
          </Link>
          <span className="header-title"> { reponame } </span>
        </div>
      )
    }
  }

  render() {
    const reponame = this.props.reponame;
    return (
      <div className="header content">
        { this.renderHeader(reponame) }
      </div>
    )
  }
}

export default Header