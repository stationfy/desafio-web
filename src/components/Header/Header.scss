.header {
  background-color: #343438;
  font-size: 1.2em;
  padding: 10px 50px;
  color: #ffffff;
  border-left: 2px solid #343438;
  border-right: 2px solid #343438;
}

.header-title-block {
  margin-left: 15px;
  padding: 35px 0px 5px;
}

.header-title {
  margin-left: 5px;
}

.header-back {
  color: #ffffff;
}
