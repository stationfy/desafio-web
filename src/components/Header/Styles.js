import styled from "styled-components";

export const Container = styled.header`
  width: 100%;
  display: flex;
  height: 85px;
  padding: 20px;
  align-items: center;
  justify-content: space-between;
  background-color: #333;
  color: #fff;

  h1 {
    margin: 0 auto;
  }

  svg {
    cursor: pointer;
  }
`;
