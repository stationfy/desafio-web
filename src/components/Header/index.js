import styled from "styled-components";

const Header = styled.form`
  padding: 0% 15%;
`;

export default Header;