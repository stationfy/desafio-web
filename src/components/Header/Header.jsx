import React from "react";
import { withRouter } from "react-router-dom";

import PropTypes from "prop-types";

import ArrowIcon from "../../assets/arrow.png";
import { Container, Arrow, BurguerMenu, HeaderContainer } from "./Styles";

const Header = props => {
  const goHome = () => props.history.push("/");
  return (
    <Container>
      {props.isPull ? (
        <Arrow src={ArrowIcon} alt="Voltar" onClick={goHome} />
      ) : (
        <BurguerMenu>
          <div></div>
          <div></div>
          <div></div>
        </BurguerMenu>
      )}
      <HeaderContainer>
        <h1>{props.children}</h1>
      </HeaderContainer>
    </Container>
  );
};

Header.propTypes = {
  children: PropTypes.string.isRequired,
  history: PropTypes.shape({
    push: PropTypes.func
  })
};

export default withRouter(Header);
