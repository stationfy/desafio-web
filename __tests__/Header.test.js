import React from 'react';
import { shallow } from 'enzyme';
import Header from '../src/components/header/Header';

describe('Testing Header Component', () => {
    it('should render correctly', () => {
        const wrapper = shallow(
            <Header title="Teste" />
        );

        expect(wrapper).toMatchSnapshot();
        wrapper.setProps({ title: "Testando" });
        expect(wrapper).toMatchSnapshot();
    });
});