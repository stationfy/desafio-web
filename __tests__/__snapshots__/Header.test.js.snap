// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Header Component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    title="Teste"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <styled.div>
        <styled.div>
                <MdMenu
                        size="20"
                />
        </styled.div>
        <styled.span>
                Teste
        </styled.span>
</styled.div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.div>
            <MdMenu
                        size="20"
            />
</styled.div>,
          <styled.span>
            Teste
</styled.span>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <MdMenu
              size="20"
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "size": "20",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Teste",
          },
          "ref": null,
          "rendered": "Teste",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <styled.div>
          <styled.div>
                    <MdMenu
                              size="20"
                    />
          </styled.div>
          <styled.span>
                    Teste
          </styled.span>
</styled.div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.div>
              <MdMenu
                            size="20"
              />
</styled.div>,
            <styled.span>
              Teste
</styled.span>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <MdMenu
                size="20"
/>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "size": "20",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Teste",
            },
            "ref": null,
            "rendered": "Teste",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Testing Header Component should render correctly 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    title="Testando"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <styled.div>
        <styled.div>
                <MdMenu
                        size="20"
                />
        </styled.div>
        <styled.span>
                Testando
        </styled.span>
</styled.div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.div>
            <MdMenu
                        size="20"
            />
</styled.div>,
          <styled.span>
            Testando
</styled.span>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <MdMenu
              size="20"
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "size": "20",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Testando",
          },
          "ref": null,
          "rendered": "Testando",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <styled.div>
          <styled.div>
                    <MdMenu
                              size="20"
                    />
          </styled.div>
          <styled.span>
                    Testando
          </styled.span>
</styled.div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.div>
              <MdMenu
                            size="20"
              />
</styled.div>,
            <styled.span>
              Testando
</styled.span>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <MdMenu
                size="20"
/>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "size": "20",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Testando",
            },
            "ref": null,
            "rendered": "Testando",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
